### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by rkdud007**
#28 

- about the test, right now i would suggest we make each unit test per each system, and later if contracts become too complicated, make tests.cairo or test folder for integration : reference : https://github.com/rkdud007/chess-dojo/blob/main/src/tests.cairo

### CODE REVIEW

**Comment by rkdud007** [comment]
I dont think 'request for review' button is active :/ @LucasLvy could you review this ?

File: contracts/src/systems/place_card.cairo [hunk 1]
```diff
@@ -1,20 +1,6 @@
...
-    defense: u8,
-    /// Energy cost of the card.
-    cost: u8,
-}
-
 #[system]
-mod place_card {
-    use super::Card;
+mod place_card_system {
+    use tsubasa::components::{Card};
```
Comment (LucasLvy) [inline contracts/src/systems/place_card.cairo:3]:
```suggestion
    use tsubasa::components::Card;
```

**Review by LucasLvy — CHANGES_REQUESTED** [review]

File: contracts/src/systems/place_card.cairo (hunk 1 above)
Comment (rkdud007) [reply contracts/src/systems/place_card.cairo]: updated

**Review by LucasLvy — APPROVED** [review]
Lgtm

**Comment by rkdud007** [comment]
@LucasLvy we need to have latest nightly version for CI, currently I have CI issue for print trait, which latest dojo solve it, and because of CI issue cannot merge it :) ( checked sozo test is working ) , suggest maybe we can merge without CI approval so this time and fix it other issue?

File: contracts/src/components.cairo [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+impl PlayerSerdeLen of dojo::SerdeLen<Option<Outcome>> {
+    #[inline(always)]
+    fn len() -> usize {
+        // 1 (variant id size) + 1 (value contained by the variant)
+        2
+    }
+}
+
+impl OutcomePrint of debug::PrintTrait<Option<Outcome>> {
```
Comment (LucasLvy) [inline contracts/src/components.cairo:50]:
```suggestion
#[cfg(test)]
impl OutcomePrint of debug::PrintTrait<Option<Outcome>> {
```

**Review by LucasLvy — CHANGES_REQUESTED** [review]
Quick change otherwise lgtm

**Review by LucasLvy — APPROVED** [review]

**Pull request merged**